src/test/java/com/fasterxml/jackson/databind/struct/UnwrapSingleArrayMiscTest.java

```diff
@@ -78,4 +81,21 @@ public void testEnumMapUnwrapping() throws Exception
             verifyException(e, "more than one value");
         }
     }
+
+    @Test
+    public void testDeserializeArrayWithNullElement() throws IOException {
+        String json = "{\"value\": [null]}";
+
+        ObjectReader r = UNWRAPPING_MAPPER.readerFor(StringWrapper.class);
+        JsonFactory factory = new JsonFactory();
+        JsonParser p = factory.createParser(json);
+        StringWrapper v = r.readValue(p);
+        assertNotNull(v);
```

Comment:
Seperating lines would help making identifying expected behavior and execution easier